src/Swarrot/Processor/Doctrine/ConnectionProcessor.php

```diff
@@ -68,16 +69,23 @@ public function process(Message $message, array $options): bool
         } finally {
             if ($options['doctrine_close_master']) {
                 foreach ($this->connections as $connection) {
-                    if ($connection instanceof MasterSlaveConnection
-                        && $connection->isConnectedToMaster()
-                    ) {
+                    if ($this->isConnectedToPrimary($connection)) {
                         $connection->close();
                     }
                 }
             }
         }
     }
 
+    private function isConnectedToPrimary(Connection $connection): bool
+    {
+        if (!class_exists('\Doctrine\DBAL\Connections\PrimaryReadReplicaConnection')) {
```

Comment:
this is not needed. You can totally do several instanceof checks (and this will be more efficient than triggering the autoloading with a `class_exists` check)